src/core/client/stream/tabs/Profile/ChangeUsername/ChangeUsernameContainer.tsx

```diff
@@ -0,0 +1,327 @@
+import { useCoralContext } from "coral-framework/lib/bootstrap";
+import { FORM_ERROR, FormApi } from "final-form";
+import { Localized } from "fluent-react/compat";
+import React, {
+  FunctionComponent,
+  useCallback,
+  useMemo,
+  useState,
+} from "react";
+import { Field, Form } from "react-final-form";
+
+import { InvalidRequestError } from "coral-framework/lib/errors";
+import { graphql, useMutation } from "coral-framework/lib/relay";
+import { withFragmentContainer } from "coral-framework/lib/relay";
+import {
+  composeValidators,
+  required,
+  validateUsername,
+  validateUsernameEquals,
+} from "coral-framework/lib/validation";
+import {
+  Box,
+  Button,
+  ButtonIcon,
+  CallOut,
+  CardCloseButton,
+  Flex,
+  FormField,
+  HorizontalGutter,
+  Icon,
+  InputLabel,
+  TextField,
+  Typography,
+} from "coral-ui/components";
+
+import { ValidationMessage } from "coral-framework/lib/form";
+
+import { ChangeUsernameContainer_viewer as ViewerData } from "coral-stream/__generated__/ChangeUsernameContainer_viewer.graphql";
+import UpdateUsernameMutation from "./UpdateUsernameMutation";
+
+import styles from "./ChangeUsername.css";
+
+interface Props {
+  viewer: ViewerData;
+}
+
+interface FormProps {
+  username: string;
+  usernameConfirm: string;
+}
+
+const ChangeUsernameContainer: FunctionComponent<Props> = ({ viewer }) => {
+  const [showEditForm, setShowEditForm] = useState(false);
+  const [showSuccessMessage, setShowSuccessMessage] = useState(false);
+  const toggleEditForm = useCallback(() => {
+    setShowEditForm(!showEditForm);
+  }, [setShowEditForm, showEditForm]);
+  const updateUsername = useMutation(UpdateUsernameMutation);
+
+  const closeSuccessMessage = useCallback(() => setShowSuccessMessage(false), [
+    setShowEditForm,
+  ]);
+
+  const canChangeUsername = useMemo(() => {
+    const { username } = viewer.status;
+    if (username && username.history.length > 1) {
+      const lastUsernameEditAllowed = new Date();
+      const dateDiff = lastUsernameEditAllowed.getDate() - 14;
+      lastUsernameEditAllowed.setDate(dateDiff);
+      return username.history.find(history => {
+        return history.createdAt > lastUsernameEditAllowed;
+      });
```

Comment:
I think it would be cleaner here to instead just reference the last history item in the list (as history items are always pushed) instead of iterating through all the history items.